BeVolt/Drivers/inc/EEPROM.h

```diff
@@ -10,29 +10,82 @@
 
 #include <stdint.h>
 
-#define EEPROM_ADDRESS 0x50 << 1
+#define EEPROM_ADDRESS (0xA0)
 
-#define EEPROM_FAULT_CODE_ADDR 0x01
-#define EEPROM_TEMP_FAULT  		 0x10
-#define EEPROM_VOLT_FAULT		   0x20
-#define EEPROM_CURRENT_FAULT	 0x30
-#define EEPROM_WWDG_FAULT		   0x40
-#define EEPROM_CAN_FAULT			 0x50
+//error codes
+#define FAULT_HIGH_TEMP			0x01
+#define FAULT_HIGH_VOLT			0x02
+#define FAULT_LOW_VOLT			0x04
+#define FAULT_HIGH_CURRENT	0x08
+#define FAULT_WATCHDOG			0x10
+#define FAULT_CAN_BUS				0x20
+
+#define EEPROM_TERMINATOR		0xFF
+
+#define EEPROM_FAULT_PTR_LOC    0x1000
+#define EEPROM_TEMP_PTR_LOC     0x1002
+#define EEPROM_VOLT_PTR_LOC	    0x1004
+#define EEPROM_CURRENT_PTR_LOC  0x1006
+#define EEPROM_WATCHDOG_PTR_LOC 0x1008
+#define EEPROM_CAN_PTR_LOC		  0x100A
 
 /** EEPROM_Init
  * Initializes I2C to communicate with EEPROM (M24128)
  */
 void EEPROM_Init(void);
 
+/** EEPROM_Load
+ * Loads the pointers from memory. Should be done after
+ * initialization and before reading or writing to the
+ * EEPROM.
+ */
+void EEPROM_Load(void);
+
+/** EEPROM_Reset
+ * Resets the EEPROM. This will clear all data that has
+ * been logged so far.
+ *
+ * This is required for clearing data and setting up new
+ * EEPROMs. This cmd really should be reserved for specialized
+ * set-up programs and the cli.
+ */
+void EEPROM_Reset(void);
+
+/** EEPROM_LogData
+ * This logs the given data at an appropriate spot in memory.
+ * This does not log an actual error, but allows more specific
+ * information about failed modules to be logged.
```

Comment:
Add attribute tags for parameters and return statements so we can generate doxygen documentation (similar to javadocs) in the future.
`@param logType : type of fault
@param data : stored into EEPROM`